code-challenges.test.js

```diff
@@ -11,42 +11,130 @@
+    expect(ourResult).toEqual(givenExpectedOutput)
+   })
+})
+    // b) Create the function that makes the test pass.
+    const filterTheWords = (array, letter) => {
+        return array.filter((word) => word.includes(letter))
+    }
 
 // --------------------3) Create a function that takes in an array of 5 numbers and determines whether or not the array is a "full house". A full house is exactly one pair and one three of a kind.
+    // Pseudocode:
+        // Create a function named 'isAFullHouse' that takes in an array of numbers and returns true is the hand is a fullhouse and it will return false if it is not
+        // Create two sets of test where the first test checks to see if the function with the given arrays are hands with a fullhouse
+        // The second test will check the given arrays with a parameter set to false for a hand that is not a fullhouse
+        // Create a passing test that sorts the hand in order using .sort()
+        // Create an object called counts
+        // Create a "for" loop that iterates over each card in the hand, checking the value of each card
+        // Create an array called values that counts the value of each card in the hand
+        // Use the Object.value method to get the vaues of the 'counts' object.
+        // Use .length to check if the length of the values array is 2 because there are only 2 unique card values in the hand
+        // If it === to 3 instead of 2, the function will return true. If not, it will return false
+        // Get expected output
+        // I truly struggled with this one, and I am not sure if I used the right technique to get an answer
 
 // a) Create a test with expect statements using the variable provided.
 
-const hand1 = [5, 5, 5, 3, 3]
-// Expected output: true
-const hand2 = [5, 5, 3, 3, 4]
-// Expected output: false
-const hand3 = [5, 5, 5, 5, 4]
-// Expected output: false
-const hand4 = [7, 2, 7, 2, 7]
-// Expected output: true
-
+describe("isAFullHouse", () => {
+    it("returns true for a fullhouse")
+        const hand1 = [5, 5, 5, 3, 3]
+        const firstResult = isAFullHouse(hand1)
+        expect(firstResult).toEqual(true)
+
+        const hand4 = [7, 2, 7, 2, 7]
+        const fourthResult = isAFullHouse(hand4)
+        expect(fourthResult).toEqual(true)
+})
+    it("returns false for a hand that is not a fullhouse", () => {
+        const hand3 = [5, 5, 5, 5, 4]
+        const thirdResult = isAFullHouse(hand3)
+        expect(thirdResult).toEqual(false)
+
+        const hand2 = [5, 5, 3, 3, 4]
+        const secondResult = isAFullHouse(hand2)
+        expect(secondResult).toEqual(false)
+    })
 // b) Create the function that makes the test pass.
+const isAFullHouse = (hand) => {
+    hand.sort((a, b) => a - b);
+    const counts = {};
+    for (const card of hand) {
+      counts[card] = (counts[card] || 0) + 1;
+    }
+    const values = Object.values(counts);
+    return values.length === 2 && (values[0] === 2 || values[0] === 3);
+  }
```

Comment:
Sweet! Nice job of keeping track using an object! Quick note: You don't need to use `.sort()` at all here, so line 133 can be removed entirely!